### PR TITLE
Adding recordings for ASM database integration tests

### PR DESCRIPTION
**Pull request by vishrutshah**
This changes is part of the feature #291
This also fixes following two issues in the ASM database SDK
1. Resetting database password was hitting the malformed URL because 'reset_password' method of sql_database_management_service.rb file at line 99 was concatenating string instead of calling method
2. When invalid location was given to create database, SDK was throwing 'No method found' in serialization.rb at line 60 while de-serializing response_xml.css('ServerName').first['FullyQualifiedDomainName'] because response_xml.css('ServerName') was not an array

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@vishrutshah__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: service_management/azure/test/integration/test_helper.rb [hunk 1]
```diff
@@ -35,7 +35,7 @@
   config.default_cassette_options = {:record => :once}
   config.cassette_library_dir = "vcr/vcr_cassettes"
   config.hook_into :faraday
-  config.allow_http_connections_when_no_cassette = false
+  config.allow_http_connections_when_no_cassette = true
```
Comment (veronicagg) [inline service_management/azure/test/integration/test_helper.rb]: do we need this, or should we just delete this line?

**Comment by veronicagg** [comment]
@vishrutshah just made one more comment, rest looks good to me. Thanks!

File: service_management/azure/test/integration/test_helper.rb (hunk 1 above)
Comment (vishrutshah) [inline service_management/azure/test/integration/test_helper.rb]: I believe that we need this line to make sure recorded tests never makes requests on network. We need this to be false. Thanks for catching this Fixing it on merge.

**Comment by vishrutshah** [comment]
@veronicagg I've fixed the lazy initialization pattern and disables VCR to make network requests on no cassette. 
Thanks for the review !

**Pull request merged**